app/Http/Controllers/Sponsors.php

```diff
@@ -152,6 +156,64 @@ private function getSponsors() {
         }
     }
 
+    private function getSponsorDisplay() {
+        // I also add on the SponsorDetail that includes the logo URL + description.
+        if(Auth::check() && in_array(Auth::user()->type, ["admin", "committee","sponsor","hacker"])) {
+            $sponsors = DB::table('sponsors')
+                ->leftJoin('sponsor_details',function($join) {
+                    $desired_type = "portal-info";
+                    $join->on('sponsors.id','=','sponsor_details.sponsor_id')
+                        ->where('sponsor_details.type','=',$desired_type);
+                })
+                ->select('sponsors.id','name','tier','payload','slug')
+                ->get();
+            return response()->json([
+                "success" => true,
+                "data" => $sponsors
+            ]);
+        } else {
+            $this->fail("Unauthorised/unauthenticated.");
+        }
+    }
+
+    private function array_flatten($array) {
+        $return = array();
+        foreach ($array as $key => $value) {
+            if (is_array($value)){ $return = array_merge($return, array_flatten($value));}
+            else {$return[$key] = $value;}
+        }
+        return $return;
+     
+     }
```

Comment:
This seems to be copy-pasted from somewhere and I think using some built-in PHP functionalities would be better:

```
function multidim_array_flatten($array){
     $it = new RecursiveIteratorIterator(new RecursiveArrayIterator($array));
     return iterator_to_array($it, true);
}
```

See https://stackoverflow.com/questions/1319903/how-to-flatten-a-multidimensional-array